kalliope/signals/geolocation/geolocation.py

```diff
@@ -0,0 +1,59 @@
+import logging
+from threading import Thread
+
+from kalliope.core import Utils
+from kalliope.core.ConfigurationManager import BrainLoader
+
+logging.basicConfig()
+logger = logging.getLogger("kalliope")
+
+
+class MissingParameter(Exception):
+    """
+    A geolocation must contain latitude, longitude, radius.
+
+    .. seealso:: Geolocation
+    """
+    pass
+
+
+class Geolocation(Thread):
+
+    def __init__(self):
+        super(Geolocation, self).__init__()
+        Utils.print_info('Init Geolocation')
+        self.brain = BrainLoader().get_brain()
+
+    def run(self):
+        logger.debug("[Geolocalisation] Loading ...")
+        self.list_synapses_with_geolocalion = self._get_list_synapse_with_geolocation(brain=self.brain)
+
+    @classmethod
+    def _get_list_synapse_with_geolocation(cls, brain):
+        """
+        return the list of synapse that use geolocation as signal in the provided brain
+        :param brain: Brain object that contain all synapses loaded
+        :type brain: Brain
+        :return: generator of synapse that use geolocation as signal
+        """
+        for synapse in brain.synapses:
+            for signal in synapse.signals:
+                # if the signal is an event we add it to the task list
+                if signal.name == "geolocation":
```

Comment:
We did this for each synapse. We shouldn't do it in the mother class and only send synapses that are related to the current processed signal?
Here we allow community signals to interact with synapses which dont' belong to the signal.